apps/namadillo/src/hooks/useExtensionEvents.tsx

```diff
@@ -26,4 +26,8 @@ export const useExtensionEvents = (): void => {
       (await injectedNamada?.isConnected()) ? "connected" : "idle"
     );
   });
+
+  useEventListenerOnce(Events.ExtensionLocked, async () => {
+    setNamadaExtensionConnected("idle");
+  });
 };
```

Comment:
One thing that is still not working for me is when you have connected to the extension before and unlock the extension. Namadillo still appears like it is disconnected, until you hit the button to connect. I think it would be desirable to have another event here to check when the extension gets unlocked.

Reply:
@pedrorezende Thanks for checking this! I ran into the same thing. The extension `App` doesn't fire any event when it's directly unlocked (that behavior only happens on connect). I will take a look and see if it's relatively do-able. Basically we want the Namadillo to re-check if it's connected to the Keychain on `App`->`Login`